cpp/src/strings/regex/regcomp.cpp

```diff
@@ -246,11 +249,50 @@ class regex_parser {
     return {false, c};
   }
 
+  // for \d and \D
+  void add_ascii_digit_class(std::vector<reclass_range>& ranges, bool negated = false)
+  {
+    if (!negated) {
+      ranges.push_back({'0', '9'});
+    } else {
+      ranges.push_back({0, '0' - 1});
+      ranges.push_back({'9' + 1, MAX_REGEX_CHAR});
+    }
+  }
+
+  // for \s and \S
+  void add_ascii_space_class(std::vector<reclass_range>& ranges, bool negated = false)
+  {
+    if (!negated) {
+      ranges.push_back({'\t', ' '});
+    } else {
+      ranges.push_back({0, '\t' - 1});
+      ranges.push_back({' ' + 1, MAX_REGEX_CHAR});
+    }
+  }
+
+  // for \w and \W
+  void add_ascii_word_class(std::vector<reclass_range>& ranges, bool negated = false)
+  {
+    if (!negated) {
+      ranges.push_back({'a', 'z'});
+      ranges.push_back({'A', 'Z'});
+      ranges.push_back({'_', '_'});
+      add_ascii_digit_class(ranges, negated);
```

Comment:
In your description you say \w maps to `[a-zA-Z_]` but this looks to be `[a-zA-Z0-9_]` which one is correct?

Reply:
The code is right. The description is wrong.

Reply:
Fixed the PR description.